hdf/src/hblocks.c

```diff
@@ -1006,14 +998,11 @@ HLIgetlink(int32 file_id, uint16 ref, int32 number_blocks)
 
 done:
     if (ret_value == NULL) { /* Error condition cleanup */
-        if (new_link->block_list != NULL)
-            HDfree(new_link->block_list);
-        if (new_link != NULL)
-            HDfree(new_link);
+        free(new_link->block_list);
```

Comment:
is new_link guaranteed to be valid after the "done"?

Reply:
Right, new_link needs to be checked for NULL first in case allocation failed.